File: Data Structures & Algorithms/missing-number/submission-2.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int xorAll = 0;
        int xorArr = 0;
        for(int i=0;i<=n;i++){
            xorAll=xorAll^i;
        }
        for(int i=0;i<n;i++){
            xorArr = xorArr^nums[i];
        }
        return xorArr^xorAll;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int mapN[100000]={0};
        int mapP[100000]={0};
        for(int i=0;i<nums.size(); i++){
            if(nums[i]<0) {
                if (mapN[nums[i]]>0)
                    return true;

                mapN[nums[i]]++;
            } else {
                if (mapP[nums[i]]>0)
                    return true;

                mapP[nums[i]]++;
            }
        }
        return false;
    }
};
